admin/services/check-window.service.js

```diff
@@ -100,6 +101,19 @@ const checkWindowService = {
   },
   assignFormsToWindow: async (checkWindowId, checkFormIds) => {
     return checkWindowDataService.sqlAssignFormsToWindow(checkWindowId, checkFormIds)
+  },
+  /**
+   * Check if pupil is allowed to log in
+   * @param pupilId
+   */
+
+  hasActiveCheckWindow: async (pupilId) => {
+    const currentCheck = await checkDataService.sqlFindOneForPupilLogin(pupilId)
+    const hasCheck = currentCheck && Object.keys(currentCheck).length > 0
+    if (!hasCheck) throw new Error(`There is no check record for pupil id ${pupilId}`)
+    const activeCheckWindow = await checkWindowDataService.sqlFindActiveCheckWindows(currentCheck.checkWindow_id)
```

Comment:
According to our preferred naming scheme as this function does not have a `one` in it I suspect it returns an array rather than an object.